core/src/main/scala/kafka/log/UnifiedLog.scala

```diff
@@ -572,6 +572,11 @@ class UnifiedLog(@volatile var logStartOffset: Long,
     result
   }
 
+  def hasOngoingTransaction(producerId: Long): Boolean = lock synchronized {
+    val entry = producerStateManager.activeProducers.get(producerId)
+    entry != null && entry.currentTxnFirstOffset.isPresent
```

Comment:
producerStateManager.activeProducers.get(producerId).exists(entry => entry.currentTxnFirstOffset.isPresent)

Reply:
This is a Java map, so that doesn't work. I can convert to scala, but not sure that is much better.